### PR TITLE
migrate common part from https://github.com/apache/servicecomb-servic…

### PR DESCRIPTION
**Pull request by tianxiaoliang**
migrate common part from https://github.com/apache/servicecomb-service-center/tree/master/pkg/rbacframe

sc and kie can leverage same code，and kie should not depend on sc code

### CODE REVIEW

**Review by little-cui — APPROVED** [review]

File: rbac/error.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"errors"
+)
+
+var (
+	ErrInvalidHeader = errors.New("invalid auth header")
+	ErrNoHeader      = errors.New("should provide Authorization header")
+	ErrInvalidCtx    = errors.New("invalid context")
+
+	ErrConvertErr = errors.New(MsgConvertErr)
+	MsgConvertErr = "type convert error"
```
Comment (AngLi2) [inline rbac/error.go:30]: Why not ErrTypeConvert = errors.New("type convert error") directly? ErrXXXErr or MsgXXXErr is confusing.

File: rbac/api.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	roles := m[ClaimsRoles]
+	roleList, err := GetRolesList(roles)
+	if err != nil {
+		return nil, ErrConvertErr
+	}
+	account := &Account{Name: a, Roles: roleList}
+	return account, nil
+}
+
+//RoleFromContext only return role name
```
Comment (AngLi2) [inline rbac/api.go:52]: whitespace is needed after //

File: rbac/resource_map.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	}
+	return r
+}
+
+//MapResource save the mapping from api to resource
+func MapResource(api, resource string) {
+	resourceMap[api] = resource
+}
+
+//BuildResourceList join the resource to an array
```
Comment (AngLi2) [inline rbac/resource_map.go:39]: whitespace is needed after //

File: rbac/resource_map.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+func GetResource(api string) string {
+	r, ok := resourceMap[api]
+	if !ok {
+		return resourceMap["*"]
+	}
+	return r
+}
+
+//MapResource save the mapping from api to resource
```
Comment (AngLi2) [inline rbac/resource_map.go:34]: whitespace is needed after //

**Review by AngLi2 — APPROVED** [review]

**Pull request merged**